internal/controller/status.go

```diff
@@ -2,11 +2,84 @@ package controller
 
 import (
 	"context"
+	"fmt"
+	"slices"
 
 	rsctv1alpha1 "github.com/ocp-power-automation/rsct-operator/api/v1alpha1"
 	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// matchPodsStatus checks if status of all the pods in slice are same or not
+func matchPodsStatus(podsStatus []string, status string) bool {
+	for _, ps := range podsStatus {
+		if ps == status {
+			continue
+		}
+		return false
+	}
+	return true
+}
+
+// evalOperatorStatus determines operator status based on the pods status
+func evalOperatorStatus(podList *corev1.PodList) string {
+	var effectiveStatus string
+	var podsStatus []string
+	for _, pod := range podList.Items {
+		switch {
+		case pod.Status.Phase == corev1.PodPending:
+			podsStatus = append(podsStatus, "PENDING")
+		case pod.Status.Phase == corev1.PodFailed:
+			podsStatus = append(podsStatus, "FAILED")
+		case pod.Status.Phase == corev1.PodRunning:
+			podsStatus = append(podsStatus, "RUNNING")
+			continue
```

Comment:
do we need a continue statement here?

Reply:
will remove the redundant `continue` here